backend/src/routes/v1/webAde.js

```diff
@@ -0,0 +1,211 @@
+  let acronyms = [];
+  const roles = req.user.jwt.realm_access.roles;
+  let hasReadAllRole = false;
+  if (typeof roles === 'object' && roles instanceof Array) {
+    acronyms = utils.filterAppAcronymRoles(roles);
+    hasReadAllRole = roles.includes('WEBADE_CFG_READ_ALL');
+  }
+
+  if (!hasReadAllRole) {
+    const appAcronym = req.params.appAcronym;
+    if (!acronyms.includes(appAcronym)) {
+      return res.status(403).json({
+        message: `User lacks permission for '${appAcronym}' acronym`
+      });
+    }
+  }
+
+  try {
+    const response = await appConfigComponent.getAppConfig(req.params.webAdeEnv, req.params.appAcronym);
+    if (response) {
+      return res.status(200).json(response);
+    } else {
+      return res.status(404).json();
+    }
+  } catch (error) {
+    log.error(error);
+    res.status(500).json({
+      message: error.message
+    });
+    return res;
+  }
+});
+
+// fetches a list of other webade apps that are dependent on a supplied acronym in a specified env
+webAde.get('/:webAdeEnv/:appAcronym/dependencies', [
+], async (req, res) => {
+
+  // TODO: a lot of this role checking is duplicate, but as we will be moving acronym management to the DB soon all this will need to be refactored anyways
+
+  // Check for required permissions. Can only fetch cfgs for the acronyms you are associated with
+  // If the user has "WEBADE_CFG_READ_ALL" then they can get all
+  let acronyms = [];
+  const roles = req.user.jwt.realm_access.roles;
+  let hasReadAllRole = false;
+  if (typeof roles === 'object' && roles instanceof Array) {
+    acronyms = utils.filterAppAcronymRoles(roles);
+    hasReadAllRole = roles.includes('WEBADE_CFG_READ_ALL');
+  }
+
+  if (!hasReadAllRole) {
+    const appAcronym = req.params.appAcronym;
+    if (!acronyms.includes(appAcronym)) {
+      return res.status(403).json({
+        message: `User lacks permission for '${appAcronym}' acronym`
+      });
+    }
+  }
+
+  try {
+    const response = await appConfigComponent.getAppConfigs(req.params.webAdeEnv);
+    if (response) {
+      return res.status(200).json(utils.filterWebAdeDependencies(response, req.params.appAcronym));
+    } else {
+      return res.status(404).json();
+    }
+  } catch (error) {
+    log.error(error);
+    res.status(500).json({
+      message: error.message
+    });
+    return res;
+  }
+});
+
+// fetches a list of all application Preferences from the webade config list that match a search criteria in the pref name
+// and have the secure indicator set to FALSE
+webAde.get('/:webAdeEnv/preferences/insecurePrefs', [
+], async (req, res) => {
+  var searchCriteria = req.query.searchCriteria;
+
+  // Check for required permissions.
+  // If the user has "WEBADE_CFG_READ_ALL" then they can get all
+  const roles = req.user.jwt.realm_access.roles;
+  let hasReadAllRole = false;
+  if (typeof roles === 'object' && roles instanceof Array) {
+    hasReadAllRole = roles.includes('WEBADE_CFG_READ_ALL');
+  }
```

Comment:
Seeing this permission check logic multiple times in this file. It may be good to consider making this a standalone function or even a middleware step. It's ok for now since we are intending on having a large refactor to shift to the DB, but it's something to have noted in backlog.

Reply:
Yeah I have a todo somewhere else in the code about this, since it will be majorly refactored at some point, have just been replicating it for now